### PR TITLE
Adding http client support for ext_proc filter

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
Risk Level: low
Testing: n/a
Docs Changes: n/a
Release Notes: inline
Fixes:

Description:
This is to address the 2st step of https://github.com/envoyproxy/envoy/issues/35488, i.e, the ext_proc HTTP client framework.

### CODE REVIEW

**Comment by yanjunxiang-google** [comment]
@tyxia  FYI: this PR is just to lay down the foundation of the file structure of http client.  There will be follow up changes when integrating the http client with the ext_proc filter code.

File: source/extensions/filters/http/ext_proc/http_client/http_client_impl.cc [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+#include "source/common/common/enum_to_int.h"
+#include "source/common/http/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+void ExtProcHttpClient::onSuccess(const Http::AsyncClient::Request&,
```
Comment (tyxia) [inline source/extensions/filters/http/ext_proc/http_client/http_client_impl.cc:11]: I think these functions here will be updated when real implementation comes in.

File: source/extensions/filters/http/ext_proc/http_client/BUILD [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+licenses(["notice"])  # Apache 2
+
+envoy_extension_package()
+
+envoy_cc_library(
+    name = "client_base_interface",
+    hdrs = ["client_base.h"],
+    tags = ["skip_on_windows"],
+    deps = [
+        "//envoy/common:pure_lib",
```
Comment (tyxia) [inline source/extensions/filters/http/ext_proc/http_client/BUILD:16]: Do we need this?

**Review by tyxia — DISMISSED** [review]
LGTM as a skeleton to start.

Thanks!

File: source/extensions/filters/http/ext_proc/http_client/http_client_impl.cc (hunk 1 above)
Comment (yanjunxiang-google) [reply source/extensions/filters/http/ext_proc/http_client/http_client_impl.cc:11]: Right

File: source/extensions/filters/http/ext_proc/http_client/BUILD (hunk 2 above)
Comment (yanjunxiang-google) [reply source/extensions/filters/http/ext_proc/http_client/BUILD]: done

**Review by tyxia — APPROVED** [review]

**Pull request merged**